de.dlr.sc.overtarget.language.tests/src/de/dlr/sc/overtarget/language/util/TargetFileHandlerTest.xtend

```diff
@@ -0,0 +1,146 @@
+/*******************************************************************************
+ * Copyright (c) 2020 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.overtarget.language.util
+
+import org.junit.Test
+import org.eclipse.core.resources.ResourcesPlugin
+import org.eclipse.core.resources.IResource
+import java.io.ByteArrayInputStream
+import org.junit.Assert
+import org.junit.Rule
+import org.junit.rules.TemporaryFolder
+import org.eclipse.emf.common.util.URI
+import org.eclipse.emf.ecore.resource.impl.ResourceSetImpl
+import de.dlr.sc.overtarget.language.targetmodel.TargetModel
+import de.dlr.sc.overtarget.language.tests.OvertargetInjectorProvider
+import org.eclipse.xtext.testing.InjectWith
+import org.eclipse.xtext.testing.XtextRunner
+import org.junit.runner.RunWith
+
+@RunWith(XtextRunner)
+@InjectWith(OvertargetInjectorProvider)
+class TargetFileHandlerTest {
+	
+	val targetFileHandler = new TargetFileHandler
+	
+	@Rule 
+	public TemporaryFolder tempFolder = new TemporaryFolder()
+	
+	@Test
+	def void testGetTargetModel() {
+		val root = ResourcesPlugin.getWorkspace().getRoot()
+		val testProject = root.getProject("de.dlr.sc.overtarget.language.tests")
+		if (testProject.exists) {
+			testProject.delete(true, null)
+		}
+		testProject.create(null)
+		if (!testProject.isOpen()) {
+			testProject.open(null)
+		}
+		val folder = testProject.getFolder("resources")
+		folder.create(IResource.NONE, true, null)
+		
+		val tmodelFile = folder.getFile("test.tmodel")
+		val bytes = "
+			Target test {
+				
+			}".getBytes()
+		val source = new ByteArrayInputStream(bytes)
+		tmodelFile.create(source, IResource.NONE, null)
+
+		val tmodel = targetFileHandler.getTargetModel(tmodelFile, null)
+		val tmodelPath = "de.dlr.sc.overtarget.language.tests/resources/test.tmodel"
```

Comment:
Maybe add some constants as class members... Such as PROJECT_NAME and TMODEL_PATH, also using each other